### PR TITLE
chore: fix useless or missing conditions.

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
finds with Deepscan.

### CODE REVIEW

File: src/Layer/LayerUpdateStrategy.js [hunk 1]
```diff
@@ -66,7 +66,6 @@ export function chooseNextLevelToFetch(strategy, node, nodeLevel = node.level, c
             // default strategy
             case STRATEGY_MIN_NETWORK_TRAFFIC:
             default:
-                nodeLevel = failureParams ? Math.ceil((currentLevel + failureParams.targetLevel) / 2) : nodeLevel;
```
Comment (zarov) [inline src/Layer/LayerUpdateStrategy.js:69]: Are you sure this is never used ?

File: src/Provider/3dTilesProvider.js [hunk 2]
```diff
@@ -101,8 +101,9 @@ export function $3dTilesIndex(tileset, baseURL) {
 
         // getBox only use inverseTileTransform for volume.region so let's not
         // compute the inverse matrix each time
+        // Assumes that node.boundingVolume is defined if node.viewerRequestVolume is undefined
```
Comment (zarov) [inline src/Provider/3dTilesProvider.js:104]: What is the confidence here ?

**Review by zarov — COMMENTED** [review]
Thanks for noticing those !

File: src/Layer/LayerUpdateStrategy.js (hunk 1 above)
Comment (gchoqueux) [reply src/Layer/LayerUpdateStrategy.js:69]: before we test if `failureParams` is `undefined`.

Comment (zarov) [reply src/Layer/LayerUpdateStrategy.js:69]: Didn't see it, my bad !

File: src/Provider/3dTilesProvider.js (hunk 2 above)
Comment (gchoqueux) [reply src/Provider/3dTilesProvider.js:104]: I think is the specification, and after in the function `getBox` ([here](https://github.com/iTowns/itowns/blob/7d3644e9d9288523d06fbb7fc347c43974bed97a/src/Provider/3dTilesProvider.js#L115))  `node.boundingVolume` is considered defined.

Comment (zarov) [reply src/Provider/3dTilesProvider.js:104]: Ok

**Review by zarov — APPROVED** [review]

**Pull request merged**